pkg/test/k8stest/client.go

```diff
@@ -6,12 +6,15 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client/fake"
 )
 
-func NewClientBuilder() *fake.ClientBuilder {
+func NewClientBuilder() (*fake.ClientBuilder, error) {
```

Comment:
Alternatively, so that we don't have to check this error everywhere, we could have NewClientBuilder and WithScheme take a `testing.TB` as the first argument and do the require.NoError assertion inside WithScheme.

Introducing a dependency on the "testing" package should be ok since it is only used by tests.

Reply:
Good suggestion, changes are in.